src/Service/Helper/NodeTypeThumbnailFormHelper.php

```diff
@@ -0,0 +1,190 @@
+    ];
+
+    // Find image fields.
+    $this->collectImageFields('node', $node_type);
+    if (empty($this->imageFields)) {
+      $form['no_image_field'] = [
+        '#type' => 'markup',
+        '#markup' => '<div>' . t('This content type has no image field yet.') . '</div>',
+      ];
+      return $form;
+    }
+
+    // Find image styles.
+    $image_styles = image_style_options(FALSE);
+    if (empty($image_styles)) {
+      $form['no_image_styles'] = [
+        '#type' => 'markup',
+        '#markup' => '<div>' . t('This site has no image style yet. Please define an image style first.') . '</div>',
+      ];
+      return $form;
+    }
+
+    $thumbnail_settings = $this->settings->get('thumbnail');
+    $node_type_thumbnail_settings = isset($thumbnail_settings[$node_type]) ?
+      $thumbnail_settings[$node_type] :
+      ['field' => '', 'style' => ''];
+    $form['field'] = [
+      '#type' => 'select',
+      '#title' => t('Select preview image to use for content recommendations.'),
+      '#options' => $this->imageFields,
+      '#default_value' => $node_type_thumbnail_settings['field'],
+      '#empty_option' => t('None'),
+      '#empty_value' => '',
+    ];
+    $form['style'] = [
+      '#type' => 'select',
+      '#title' => t('Select the style to use for the thumbnail.'),
+      '#options' => $image_styles,
+      '#default_value' => $node_type_thumbnail_settings['style'],
+      '#empty_option' => t('None'),
+      '#empty_value' => '',
+      '#states' => [
+        'visible' => [
+          ':input[name="acquia_lift[field]"]' => ['!value' => ''],
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Traverse the FieldableEntity and its fields, collect a field "roadmap" that
+   * can lead to a image file.
+   *
+   * @param string $target_type
+   *   Fieldable entity's identifier.
+   * @param string $type
+   *   Type of the fieldable entity.
+   * @param string $key_prefix
+   *   The concatenated entity field keys that has been traversed through.
+   * @param string $label_prefix
+   *   The concatenated entity labels that has been traversed through.
+   */
+  private function collectImageFields($target_type, $type, $key_prefix = '', $label_prefix = '') {
+    $field_definitions = $this->entityManager->getFieldDefinitions($target_type, $type);
+    foreach ($field_definitions as $field_key => $field_definition) {
+      $field_type = $field_definition->getType();
+      $field_target_type = $field_definition->getSetting('target_type');
+      $field_label = $field_definition->getLabel();
+      $full_label = $label_prefix . $field_label;
+      $full_key = $key_prefix . $field_key;
+
+      // Keep this field, if the field has already been processed.
+      $field_hash = spl_object_hash($field_definition);
+      if (isset($this->processedFieldHashes[$field_hash])) {
+        continue;
+      }
+      $this->processedFieldHashes[$field_hash] = TRUE;
+
+      // 1) Image type.
```

Comment:
I have a recollection of supporting media/video in version 7.  Is there a way to do that here and/or maybe we just want to provide a hook to allow different media types to be supported?